nativelib/src/main/resources/time.c

```diff
@@ -30,3 +30,28 @@ long long scalanative_nano_time() {
 
   return nano_time;
 }
+
+long long scalanative_current_time_millis() {
+	long long current_time_millis;
+
+#define NANOS_PER_MILLI 1000000
+#define MILLIS_PER_SEC 1000
+#define MICROS_PER_MILLI 1000
+
+#ifdef __MACH__
+	struct timeval  tv;
+	gettimeofday(&tv, NULL);
```

Comment:
I'd prefer to stick to just using `gettimeofday` for simplicity, if we see it's a problem for portability, we can revise the solution.